src/Microsoft.ML.Transforms/Text/WordEmbeddingsTransform.cs

```diff
@@ -324,9 +343,189 @@ public Mapper(WordEmbeddingsTransform parent, Schema inputSchema)
+            {
+                foreach (var (input, output) in _parent.Columns)
+                {
+                    var srcVariableName = ctx.GetVariableName(input);
+                    var schema = _parent.GetOutputSchema(InputSchema);
+                    var dstVariableName = ctx.AddIntermediateVariable(schema[output].Type, output);
+                    SaveAsOnnxCore(ctx, srcVariableName, dstVariableName);
+                }
+            }
+
+            private void SaveAsOnnxCore(OnnxContext ctx, string srcVariableName, string dstVariableName)
+            {
+                // Converts 1 column that is taken as input to the transform into one column of output
+                //
+                // Missing words are mapped to k for finding average, k + 1 for finding min, and k + 2 for finding max
+                // Those spots in the dictionary contain a vector of 0s, max floats, and min floats, respectively
+                //
+                // Symbols:
+                // j: length of latent vector of every word in the pretrained model
+                // n: length of input tensor (number of words)
+                // X: word input, a tensor with n elements.
+                // k: # of words in pretrained model (known when transform is created)
+                // S: word labels, k tensor (known when transform is created)
+                // D: word embeddings, (k + 3)-by-j tensor(known when transform is created). The extra three embeddings
+                //      at the end are used for out of vocab words.
+                // F: location value representing missing words, equal to k
+                // P: output, a j * 3 tensor
+                //
+                //                                                      X [n]
+                //                                                       |
+                //                           LabelEncoder (classes_strings = S [k], default_int64 = k)
+                //                           /   |                       |                           \
+                //                          /    | Initialize (F) ---> Equal                          \
+                //                         /     |                     / |  \                          \
+                //                        /      '-------------|      /  |   \                          \
+                //      -----------------/               ------|-----/   |    \------------------        \----------
+                //     /                                /      |         |                       \                  \
+                //    |                     Cast (to = int64)  | Cast (to = float)              Not                  |
+                //    |                            |           |         |                        |                  |
+                //    '------------ Add -----------'           | Scale (scale = 2.0)         Cast (to = int32)       |
```

Comment:
Missing variable names between operators!

Reply:
Thanks, should be good now!

---
In reply to: [227994103](https://github.com/dotnet/machinelearning/pull/1249#discussion_r227994103) [](ancestors = 227994103)